core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -539,6 +539,13 @@ default void timeoutNow() {
      */
     void mutateAdditionalRequestHeaders(Consumer<HttpHeadersBuilder> mutator);
 
+    /**
+     * Disconnects gracefully connection and returns {@link CompletableFuture} that completes when the channel
+     * is closed.
+     */
+    @UnstableApi
+    CompletableFuture<Void> disconnectWhenFinished();
```

Comment:
Could you rename it to `initiateConnectionShutdown()` and move it up to `RequestContext`, so that `ClientRequestContext` and `ServiceRequestContext` share the same method?

Reply:
After I rename to 'initiateConnectionShutdown', I need `InitiateConnectionShutdown` event without drain duration.
What it good to use 'InitiateConnectionShutdown' and ignore drain duration or to create new event class?